src/Sentry.Unity.Android/SentryJava.cs

```diff
@@ -57,5 +58,103 @@ public static void Close()
+                gpu.SetIfNotNull("memorySize", GpuMemorySize);
+                gpu.SetIfNotNull("apiType", GpuApiType);
+                gpu.SetIfNotNull("multiThreadedRendering", GpuMultiThreadedRendering);
+                gpu.SetIfNotNull("version", GpuVersion);
+                gpu.SetIfNotNull("npotSupport", GpuNpotSupport);
+
+                using var sentry = GetSentryJava();
+                sentry.CallStatic("configureScope", new ScopeCallback(scope =>
+                {
+                    using var contexts = scope.Call<AndroidJavaObject>("getContexts");
+                    contexts.Call("setGpu", gpu);
+                }));
+            }
+        }
+
+        private static void SetIfNotNull<T>(this AndroidJavaObject javaObject, string property, T? value, string? valueClass = null)
+        {
+            if (value is not null)
+            {
+                if (valueClass is null)
+                {
+                    javaObject.Set(property, value!);
+                }
+                else
+                {
+                    using var valueObject = new AndroidJavaObject(valueClass, value!);
+                    javaObject.Set(property, valueObject);
+                }
+            }
+        }
+
+        private static void SetIfNotNull(this AndroidJavaObject javaObject, string property, int? value) =>
+            SetIfNotNull(javaObject, property, value, "java.lang.Integer");
+
+        private static void SetIfNotNull(this AndroidJavaObject javaObject, string property, bool? value) =>
+            SetIfNotNull(javaObject, property, value, "java.lang.Boolean");
+
+        /// Implements the io.sentry.ScopeCallback interface.
+        internal class ScopeCallback : AndroidJavaProxy
+        {
+            private readonly Action<AndroidJavaObject> _callback;
+
+            public ScopeCallback(Action<AndroidJavaObject> callback) : base("io.sentry.ScopeCallback")
+            {
+                _callback = callback;
+            }
+
+            // Note: defining the method should be enough with the default Invoke(), but in reality it doesn't work:
+            // No such proxy method: Sentry.Unity.Android.SentryJava+ScopeCallback.run(UnityEngine.AndroidJavaObject)
+            //   public void run(AndroidJavaObject scope) => UnityEngine.Debug.Log("run() invoked");
+            // Therefore, we're overriding the Invoke() instead:
+            public override AndroidJavaObject? Invoke(string methodName, AndroidJavaObject[] args)
+            {
+                try
+                {
+                    if (methodName != "run" || args.Length != 1)
+                    {
+                        throw new Exception($"Invalid invocation: {methodName}({args.Length} args)");
+                    }
+                    _callback(args[0]);
+                }
+                catch (Exception e)
+                {
+                    UnityEngine.Debug.LogError($"Error in SentryJava.ScopeCallback: {e}");
```

Comment:
We have a policy to avoid sending SDK errors as events (and charging our customers for it). We handle it and log through the diag logger, and users must go on debug mode to find out. It's a challenge of crash reporting services but folks in general are not happy when we're erroring out and causing their apps to break or burnign through their quota and asking them for money when it's our fault